arch/arm/src/cxd56xx/cxd56_emmc.c

```diff
@@ -440,7 +440,17 @@ static void emmc_send(int datatype, uint32_t opcode, uint32_t arg,
 
   /* Wait for command or data transfer done */
 
-  emmc_takesem(&g_waitsem);
+  do
+    {
+      ret = emmc_takesem(&g_waitsem);
+
+      /* The only expected error is ECANCELED which would occur if the
+       * calling thread were canceled.
+       */
+
+      DEBUGASSERT(ret == OK || ret == -ECANCELED);
+    }
+  while (ret < 0);
```

Comment:
This will lose the cancelation error.  The function will return OK and the cancelation will not occur.  I recommend you consider return ret at the bottom of the file instead of OK.

Perhaps that ret value is  destroyed by later logic??? I can't see that here.  If that is the case then you should save any error returned by emmc_taksem() in some other variable and return that.